### PR TITLE
CollapsedBroadcastsService

### PR DESCRIPTION
**Pull request by tatitati**
- New test class for method: countUpcomingByProgramme()
- Added additional comments in original service (too much?, is it OK?)
- Added additional tests for some cases (boundaries)

### CODE REVIEW

File: tests/Service/CollapsedBroadcastsService/AbstractCollapsedBroadcastServiceTest.php [hunk 1]
```diff
@@ -36,4 +43,18 @@ protected function service(): CollapsedBroadcastsService
             $this->mockServiceRepository
         );
     }
+
+    /**
+     * @param Category[] $categories
+     * @return int[]
+     */
+    protected function extractDbAncestryIds(array $categories): array
```
Comment (BPScott) [inline tests/Service/CollapsedBroadcastsService/AbstractCollapsedBroadcastServiceTest.php]: This is only used in one test class, is it worth putting it in the abstract test case?

File: tests/Service/CollapsedBroadcastsService/FilterCategoriesByBroadcastedDateTest.php [hunk 2]
```diff
@@ -1,40 +1,71 @@
...
-        $this->assertCount(1, $dbBroadcastedCategories);
-        $this->assertSame([1, 2, 3], $dbBroadcastedCategories[0]->getDbAncestryIds());
+        $this->assertContainsOnly(Genre::class, $categories);
+        $this->assertEquals($expectedCategoriesFoundOnDate, $this->extractDbAncestryIds($categories));
+    }
+
+    public function categoriesBroadcastedOnDateProvider(): array
+    {
+        return [
+            'CASE: none of category specified was not broadcasted on date' => [
```
Comment (BPScott) [inline tests/Service/CollapsedBroadcastsService/FilterCategoriesByBroadcastedDateTest.php]: This seems to have the same key as the test case below

Also double negatives sound weird, is there a better way to phrase this?

File: tests/Service/CollapsedBroadcastsService/FindByCategoryAndStartAtDateRangeTest.php [hunk 3]
```diff
@@ -2,36 +2,98 @@
...
-            ->with($ancestry, false, $fromDate, $toDate)
-            ->willReturn($broadcastData);
+            ->willReturn([
+                 ['areWebcasts' => [false, false], 'serviceIds' => [111, 222], 'broadcastIds' => [1, 2, 3, 4]],
+                 ['areWebcasts' => [true, true], 'serviceIds' => [333, 444], 'broadcastIds' => [3, 4, 56, 67]],
+                 ['areWebcasts' => [true, false], 'serviceIds' => [555, 666], 'broadcastIds' => [5, 6, 100]],
+                 ['areWebcasts' => [false, false], 'serviceIds' => [false, false], 'broadcastIds' => [7, 8, 20, 48, 23]],
+                 ['areWebcasts' => [false, false], 'serviceIds' => [true, true], 'broadcastIds' => [8, 9, 12, 122]],
+                 ['areWebcasts' => [false, false], 'serviceIds' => [null, null], 'broadcastIds' => [10, 11]],
+                 ['areWebcasts' => ['0', 0], 'serviceIds' => [777, 888], 'broadcastIds' => [14, 15]],
```
Comment (BPScott) [inline tests/Service/CollapsedBroadcastsService/FindByCategoryAndStartAtDateRangeTest.php]: Whats with these return values? Can the findByCategoryAncestryAndStartAtDateRange ever return data where areWebcasts is a string or int?

File: tests/Service/CollapsedBroadcastsService/AbstractCollapsedBroadcastServiceTest.php (hunk 1 above)
Comment (tatitati) [reply tests/Service/CollapsedBroadcastsService/AbstractCollapsedBroadcastServiceTest.php]: yeah, you are right

File: tests/Service/CollapsedBroadcastsService/FilterCategoriesByBroadcastedDateTest.php (hunk 2 above)
Comment (tatitati) [reply tests/Service/CollapsedBroadcastsService/FilterCategoriesByBroadcastedDateTest.php]: Yeah!, phpstan was handy here, I just fixed it

**Comment by BPScott** [comment]
Well, approved once you fix the thing phpstan has pointed out

**Pull request merged**